### ISSUE TITLE
"Too many commas in name" errors using Better BibLaTeX

### ISSUE THREAD

**Issue by vadim0x60** [retorquere/zotero-better-bibtex#3109]
### Debug log ID

YJQ3P6D9-refs-euc/6.7.263-7

### What happened?

I export my bibliography with "Better BibLaTeX" and when I run `latexmk` I get "too many commas in name" errors for some of the citations (the ones I selected for the debug log), i.e.

`Too many commas in name 20 of "Reed, Scott and Zolna, Konrad and Parisotto, Emilio and Colmenarejo, Sergio Gomez and Novikov, Alexander and Barth-Maron, Gabriel and Gimenez, Mai and Sulsky, Yury and Kay, Jackie and Springenberg, Jost Tobias and Eccles, Tom and Bruce, Jake and Razavi, Ali and Edwards, Ashley and Heess, Nicolas and Chen, Yutian and Hadsell, Raia and Vinyals, Oriol and Bordbar, Mahyar and family=Freitas, given=Nando, prefix=de, useprefix=true" for entry reedGeneralistAgent2022 while executing---line 2815 of file biblatex.bst`

and document compilation fails. If I use Zotero's built in BibLaTeX exporter instead, these errors don't happen, everything works fine. I'd stick to the built in exporter, but it doesn't support auto-updates.

**Comment by retorquere** [comment]
@njbart if one name is in extended mode, must all names be in extended mode?

**Comment by retorquere** [comment]
Are you sure you're using biblatex for compilation BTW?

**Comment by njbart** [comment]
> If one name is in extended mode, must all names be in extended mode?

No.

> Are you sure you're using biblatex for compilation BTW?

That's a great question. A minimal working example (MWE) would have been useful. Regardless, it would be helpful to know the versions of biblatex and biber that were being used, in case they were outdated. (Though the extended name format for biblatex was introduced no later than 2016; see e.g. https://github.com/retorquere/zotero-better-bibtex/issues/566).

**Comment by retorquere** [comment]
The error mentions a .bst style file, so likely a bibtex toolchain rather dan biblatex. @vadim0x60 if you want to use a bibtex toolchain you should export using Better BibTeX rather than Better BibLaTeX.

**Comment by vadim0x60** [comment]
This is how I include the bibliography file:

```
\usepackage[style=alphabetic,citestyle=alphabetic,sorting=nyt,sortcites=true,autopunct=true,autolang=other,hyperref=true,abbreviate=false,backref=true,backend=bibtex]{biblatex}
\addbibresource{thesis.bib}
```

In order to produce `thesis.bib`, I export my bibliography using BibLaTeX (not BibTeX and not Better BibLaTeX) and it works just fine.

as for the versions:

```
> tlmgr show biblatex
...
installed:   Yes
revision:    66403
sizes:       doc: 19997k, run: 2505k
relocatable: No
cat-version: 3.19
...
```

```
> tlmgr show biber
...
installed:   Yes
revision:    68188
sizes:       src: 1233k, doc: 269k, bin: 98709k
relocatable: No
cat-version: 2.19
...
```

**Comment by retorquere** [comment]
That is because the standard bibtex does not support the extended name format. As said, you can turn off extended name format in the BBT prefs (at the cost of worse sorting in your bibliography) and you will get the same name for Better BibLaTeX.

**Comment by njbart** [comment]
> That is because the standard biblatex does not support the extended name format. 

No, that's not actually true. In fact you have to actively disable the extended name format if you don't want it (see biblatex manual, v 3.20, p 81).

The most likely culprit is `backend=bibtex`. Try `backend=biber` instead.

**Comment by retorquere** [comment]
Sorry - I meant "zotero's bundled biblatex export does not support extended name format"

**Comment by retorquere** [comment]
@vadim0x60 if you can't confirm you're using biblatex I will have to close the issue.
